File: HeartForCharity/HeartForCharity.Services/Database/OrganisationProfile.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Net;

namespace HeartForCharity.Services.Database
{
    public class OrganisationProfile
    {
        [Key]
        public int OrganisationProfileId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        public int UserId { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = null!;

        [MaxLength(2000)]
        public string? Description { get; set; }

        [MaxLength(255)]
        [EmailAddress]
        public string? ContactEmail { get; set; }

        [MaxLength(20)]
        public string? ContactPhone { get; set; }

        [ForeignKey(nameof(Address))]
        public int? AddressId { get; set; }

        [MaxLength(500)]
        public string? LogoUrl { get; set; }

        [ForeignKey(nameof(OrganisationType))]
        public int? OrganisationTypeId { get; set; }

        public bool IsVerified { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DeletedAt { get; set; }

        // Navigacijska svojstva
        public virtual User User { get; set; } = null!;
        public virtual Address? Address { get; set; }
        public virtual OrganisationType? OrganisationType { get; set; }
        public virtual ICollection<Campaign> Campaigns { get; set; } = new List<Campaign>();
        public virtual ICollection<VolunteerJob> VolunteerJobs { get; set; } = new List<VolunteerJob>();
        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/organisation_profile.dart
class OrganisationProfile {
  final int organisationProfileId;
  final int userId;
  final String name;
  final String? description;
  final String? contactEmail;
  final String? contactPhone;
  final String? logoUrl;
  final int? organisationTypeId;
  final String? organisationTypeName;
  final int? addressId;
  final String? cityName;
  final String? countryName;
  final DateTime createdAt;

  OrganisationProfile({
    required this.organisationProfileId,
    required this.userId,
    required this.name,
    this.description,
    this.contactEmail,
    this.contactPhone,
    this.logoUrl,
    this.organisationTypeId,
    this.organisationTypeName,
    this.addressId,
    this.cityName,
    this.countryName,
    required this.createdAt,
  });

  factory OrganisationProfile.fromJson(Map<String, dynamic> json) =>
      OrganisationProfile(
        organisationProfileId: json['organisationProfileId'],
        userId: json['userId'],
        name: json['name'],
        description: json['description'],
        contactEmail: json['contactEmail'],
        contactPhone: json['contactPhone'],
        logoUrl: json['logoUrl'],
        organisationTypeId: json['organisationTypeId'],
        organisationTypeName: json['organisationTypeName'],
        addressId: json['addressId'],
        cityName: json['cityName'],
        countryName: json['countryName'],
        createdAt: DateTime.parse(json['createdAt']),
      );
}
